app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -1513,12 +1516,17 @@ private void updateControllerBatteryLevels(final int leftLevel, final int rightL
     }
 
     private void updateBatterLevels(final int leftLevel, final int rightLevel) {
-        BatteryManager bm = (BatteryManager)getSystemService(BATTERY_SERVICE);
-        int battery = bm == null ? 100 : bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
+        long currentTime = System.nanoTime();
+        if (((currentTime - mLastBatteryUpdate) >= BATTERY_UPDATE_INTERVAL) || mLastBatteryLevel == -1) {
+            mLastBatteryUpdate = currentTime;
+            BatteryManager bm = (BatteryManager)getSystemService(BATTERY_SERVICE);
+            mLastBatteryLevel = bm == null ? 100 : bm.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY);
+        }
+
         Intent intent = this.registerReceiver(null, new IntentFilter(Intent.ACTION_BATTERY_CHANGED));
```

Comment:
This can be addressed in another PR, but I think that it would make sense to actually register a receiver for this broadcast so we can update the UI faster when there is a change (e.g. when the user plugs in the device).

Reply:
Hmm not sure what you mean. I'm just reducing the rate of updates of the battery level. The code bellow the one you mention is the one that detects plug/unplug and that's executed every second (as that's the rate of updates that we get from `BrowserWorld.cpp`)